plugins/search-backend-node/src/engines/LunrSearchEngine.ts

```diff
@@ -163,30 +164,38 @@ export class LunrSearchEngine implements SearchEngine {
 
     const results: LunrResultEnvelope[] = [];
 
+    const indexKeys = Object.keys(this.lunrIndices).filter(
+      type => !documentTypes || documentTypes.includes(type),
+    );
+
+    if (documentTypes?.length && !indexKeys.length) {
+      throw new MissingIndexError(
+        `Missing index for ${documentTypes?.toString()}. This means there are no documents to search through.`,
```

Comment:
Just want to check this error message.

> _This means there are no documents to search through._

If a collator indexes an empty array of documents, would this error case actually occur?

Reply:
I believe this is true for Lunr and Postgres as those doesn't return as good of a error message as elastic does. We could write a test for that use case. And if thats the case we could probably tweak the error message for those two a bit. What do you think @iamEAP? cc @camilaibs

Reply:
Yeah, maybe we back out the change in this PR to Postgres because there isn't actually a way to distinguish between "the collator never successfully ran" and "the collator successfully ran, but returned 0 results". (see other review comment)

I think we could also make the existing error messages a little clearer, though, because I think they can happen under several circumstances.  Maybe something like this?

> _Missing index for {type}. This could be because the index hasn't been created yet or there was a problem during index creation._